static/js/util/ProfileFormFields.js

```diff
@@ -64,13 +66,41 @@ export default class ProfileFormFields extends React.Component {
     }));
   }
 
+  stepBack = () => {
+    this.context.router.push(this.prevUrl);
+  };
+
   saveAndContinue = () => {
-    let lastTab = this.nextUrl === "/dashboard";
-    saveAndContinue.call(this, lastTab).then(() => {
+    saveProfileStep.call(this, this.isLastTab).then(() => {
       this.context.router.push(this.nextUrl);
     });
   };
 
+  progressControls = (cellWidth=10) => {
+    let prevButton, nextButton;
+    if(this.prevUrl) {
+      prevButton = <Button
+        raised
+        className="progress-button previous"
+        onClick={this.stepBack}>
+        <span>Previous</span>
+      </Button>;
+    }
+    if(this.nextUrl) {
+      nextButton = <Button
+        raised
+        colored
+        className="progress-button next"
+        onClick={this.saveAndContinue}>
+        <span>{this.isLastTab ? "I'm Done!" : "Save and Continue"}</span>
+      </Button>;
+    }
+    return <Cell col={cellWidth}>
+        {prevButton}
+        {nextButton}
+      </Cell>;
```

Comment:
I think the main reason we prefer methods (e.g. on `ProfileFormFields`) over separate components for a lot of stuff is that many of those methods need to pull stuff out of `this.props`, and end up being tightly coupled to profile display functionality. In general, I think if we don't see a real need to use a component elsewhere it's a little easier (and less verbose! haha) to write instance helper methods that we call in the `render` method of our component. With a separate component (such as `<ProfileProgressControls>` or something) and the current setup for `nextUrl` and so on we'd need to splat in `this.props` and also assign props for all of the instance variables we'd need, so the call would end up looking like:

```js
<Cell col={2} />
<ProfileProgressControls {...this.props} col={10} nextUrl={this.nextUrl} prevUrl={this.prevUrl} />
<Cell col={2} />
```

IMO in this specific case the helper method (instance method on `ProfileFormFields`) has less boilerplate and ends up being cleaner, but in other cases this might not be true. I think the line between separate component / helper method is a little unclear sometimes, and I had a bit of trouble pulling the work history editing component out of the work history tab (so I could display it on the `/users/` page) partially for that reason. So we should definitely always be ensuring we're being somewhat sane with this stuff - we don't want tons and tons of tiny components, but we also don't want god classes that end up having far too much responsibility and becoming difficult to refactor.